team-validator.js

```diff
@@ -253,16 +253,15 @@ class Validator {
 				if (!set.moves[i]) continue;
 				let move = tools.getMove(Tools.getString(set.moves[i]));
 				if (!move.exists) return ['"' + move.name + '" is an invalid move.'];
-				set.moves[i] = move.name;
```

Comment:
The `Pokemon` ruleset does not apply to custom games (and certain OMs). If you want a validator-level guarantee, it needs to be enforced at the validator level.